### ISSUE TITLE
Emotion with SWC does not work with next/jest (Next 12.1.1)

### ISSUE THREAD

**Issue by pgrippi** [vercel/next.js#35661]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 20.6.0: Wed Jan 12 22:22:42 PST 2022; root:xnu-7195.141.19~2/RELEASE_X86_64
Binaries:
  Node: 16.13.2
  npm: 8.1.2
  Yarn: 1.22.17
  pnpm: N/A
Relevant packages:
  next: 12.1.1
  react: 17.0.2
  react-dom: 17.0.2
```

### Describe the Bug

I've upgraded my next.js app to 12.1.1 and have enabled SWC and emotion in my next config via the `experimental: {emotion: true}` option

My `jest.config.js` file looks like this:

```js
const nextJest = require('next/jest');

const createJestConfig = nextJest({
  dir: './src',
});

const customJestConfig = {};

module.exports = createJestConfig(customJestConfig);
```

I have a component that uses the nested emotion selector that looks like this:

```js
const Results = styled.div`
  margin: 0;
  padding: 0;
`;

const Container = styled.div`
  ${Results}:first-of-type {
    border-bottom: 1px solid #000;
  }
`;
```

When I run the app using `next dev` or build with `next build` everything works, the SWC plugin is correctly able to handle the nested selectors. But when I run `jest`, it errors out with:

```
console.error
  Error: Uncaught [Error: Component selectors can only be used in conjunction with @emotion/babel-plugin.]
```

### Expected Behavior

`next/jest` correctly configures the swc emotion transform when `experimental: {emotion: true}` is set in the next config file

### To Reproduce

I've set up a simple repo based on the `with-emotion-swc` example: https://github.com/pgrippi/nextjs-emotion-swc-jest-bug

* Clone repo
* `yarn install`
* `yarn test` to see the jest failure
* `yarn dev` / `yarn build` to see that the application works

**Comment by pgrippi** [comment]
Adding a comment so that the github bot doesn't close this (it's been almost 30 days) - not looking for an update but this is still an issue in Next 12.1.5

**Comment by balazsorban44** [comment]
@pgrippi only issues marked with "please add a complete reproduction" are affected by the stale bot.

We never close acknowledged/verified bug reports, even after 30 days.

This issue is already tracked and being worked on, you can check #36500

**Comment by pgrippi** [comment]
Can confirm that this is now fixed in 12.1.6, thank you!!
